examples/preprocessing/plot_pace.py

```diff
@@ -0,0 +1,372 @@
+"""
+Functional Principal Component Analysis through Conditional Expectation
+=======================================================================
+
+Explores an alternative way to do functional principal component analysis for
+irregularly sampled data.
+"""
+
+# Author: Alejandro Arias Gomez
+# License: MIT
+
+# sphinx_gallery_thumbnail_number = 6
+
+# %%
+from collections import defaultdict
+
+import matplotlib.pyplot as plt
+import numpy as np
+from sklearn.utils import Bunch
+
+from skfda.datasets._real_datasets import fetch_growth
+from skfda.datasets._sample_from_fdata import irregular_sample
+from skfda.preprocessing.dim_reduction import FPCA, PACE
+from skfda.representation import FDataGrid
+
+# %%
+# In this example we are going to use functional principal component analysis
+# through conditional expectation to explore the Berkeley Growth Study and
+# obtain conclusions about said dataset using this technique, comparing the
+# results to those obtained with the FPCA method.
+#
+# The PACE algorithm is an alternative to FPCA that is specifically designed
+# for irregularly sampled data. It uses local linear smoothing to estimate
+# the model components of the data, and then performs the conditional
+# expectation step to obtain the principal components. These components are
+# the directions that capture the main modes of variation across the function.
+# PACE shares the same objectives as FPCA, it is a dimensionality
+# reduction method for functional data that aims to reduce the complexity of
+# studying observations by expressing the data in terms of a basis of K
+# components that explain most of the variation in the data.
+#
+# The PACE algorithm was introduced in :footcite:ts`yao+muller+wang_2005_pace`,
+# and in this example we will use the implementation of the algorithm that
+# follows the same steps as the original algorithm.
+#
+# The Berkeley Growth Study dataset consists of height measurements over time
+# for a cohort of children. It is a classic dataset in Functional Data
+# Analysis, and here it is initially provided in a dense, regular format
+# (FDataGrid).
+dataset: Bunch = fetch_growth()
+fd: FDataGrid = dataset.data
+assert isinstance(fd, FDataGrid), "Expected an FDataGrid object"
+
+fd[:20].plot()
+plt.show()
+
+# %%
+# To evaluate the performance of PACE, we transform the dense dataset into an
+# irregular one. Each subject is now sampled at 8 randomly selected time
+# points. This mirrors the sparse and irregular setting often found in
+# real-world longitudinal data.
+random_state = 12
+
+irregular_fd = irregular_sample(
+    fd,
+    8,
```

Comment:
Set a variable to this value and then use the variable here. This forces you to choose a descriptive name for the variable, which in turn makes easier for the reader to understand what that means.

Alternatively, use keyword arguments.